src/hooks/useUSDValue.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  Token,
+  Trade,
+  currencyEquals,
+  Price,
+  Currency,
+  CurrencyAmount,
+  JSBI,
+  ZERO,
+  ONE,
+  Fraction,
+  RoutablePlatform
+} from '@swapr/sdk'
+import { useTradeExactOutAllPlatforms } from './Trades'
+import { tryParseAmount } from '../state/swap/hooks'
+import { ChainId } from '@swapr/sdk'
+import { USDC, DAI } from '../constants/index'
+import { useActiveWeb3React } from './index'
+import { useMemo } from 'react'
+
+const STABLECOIN_OUT: { [chainId: number]: Token } = {
+  [ChainId.MAINNET]: DAI,
+  [ChainId.ARBITRUM_ONE]: USDC[ChainId.ARBITRUM_ONE],
+  [ChainId.XDAI]: USDC[ChainId.XDAI]
+}
+
+const ALLOWED_PLATAFORMS: { [chainId: number]: string[] } = {
+  [ChainId.MAINNET]: [RoutablePlatform.UNISWAP.name, RoutablePlatform.SUSHISWAP.name],
+  [ChainId.ARBITRUM_ONE]: [RoutablePlatform.SUSHISWAP.name],
+  [ChainId.XDAI]: [RoutablePlatform.SUSHISWAP.name]
+}
+
+const AMOUNT_OUT = '1'
+
+export function useUSDPrice(currency?: Currency, selectedTrade?: Trade) {
+  const { chainId } = useActiveWeb3React()
+  const stablecoin = chainId ? STABLECOIN_OUT[chainId] : undefined
+  const parsedAmountOut = tryParseAmount(AMOUNT_OUT, stablecoin, chainId)
+
+  const tradeExactOutAllPlatforms = useTradeExactOutAllPlatforms(currency, parsedAmountOut)
+
+  return useMemo(() => {
+    if (!currency || !chainId || !stablecoin || !tradeExactOutAllPlatforms) return undefined
+
+    if (currencyEquals(currency, stablecoin)) return new Price(currency, currency, '1', '1')
```

Comment:
This creates a technical debt - the list of stable coins continues to grow every day. We can remove it, Most DEX has stable coin pairs. On top of that, the peg between any stable coin and its fiat value is not a perfect 1:1.

Reply:
We can't simulate a trade for the same token so we are defining that: if the selected currency is the same as the one we defined as stable coin we do a 1:1 value match.

Reply:
I missed that. Makes sense now.